### PR TITLE
fix(TDI-42154): Escape char in Couchbase data

### PR DESCRIPTION
**Pull request by Voievidko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-42154

**What is the new behavior?**
I add logic to recognize valid JSON if checkbox `contains JSON` is active. And now Couchbase component can create JSON from all incoming data, even if some fields will be just Strings and others JSON

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-couchbase/src/test/java/org/talend/components/couchbase/runtime/CouchbaseWriterTest.java [hunk 1]
```diff
@@ -104,4 +101,30 @@ public void testGetWriteOperation() {
...
+        Mockito.when(writeOperation.getSink()).thenReturn(sink);
+        writer = new CouchbaseWriter(writeOperation);
+
+        // Prepare objects.
+        CouchbaseConnection connection = Mockito.mock(CouchbaseConnection.class);
+        Mockito.when(sink.getConnection()).thenReturn(connection);
+        Mockito.when(sink.getContainsJson()).thenReturn(true);
+
+        Schema schema = SchemaBuilder.builder().record("record").fields().requiredString("idFieldName").endRecord();
+        IndexedRecord record = new GenericRecordBuilder(schema).set("idFieldName", 1).build();
```
Comment (wwang-talend) [inline components/components-couchbase/src/test/java/org/talend/components/couchbase/runtime/CouchbaseWriterTest.java]: the test seems only test the basic input, not the json string input as one column value, best to do adjust if i am right.

File: components/components-couchbase/src/main/java/org/talend/components/couchbase/runtime/CouchbaseWriter.java [hunk 2]
```diff
@@ -83,13 +87,33 @@ public void write(Object datum) throws IOException {
...
+            } else {
+                connection.upsert(id.toString(), datum.toString());
+            }
             result.successCount++;
         } catch (Exception e) {
             handleException("Record is not processed. Failed to upsert value - " + datum.toString(), e);
         }
     }
 
+    private JsonObject createHierarchicalJson(Schema schema, IndexedRecord record, int idPos){
```
Comment (wwang-talend) [inline components/components-couchbase/src/main/java/org/talend/components/couchbase/runtime/CouchbaseWriter.java]: the method is the core for the new json string input, advice to do some real test about the method for all cases which sure contain the one in current jira.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/7/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/7/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arvoievidko_TDI-42154_Escape_char_in_couchbase_data)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/8/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/8/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arvoievidko_TDI-42154_Escape_char_in_couchbase_data)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/9/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/9/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arvoievidko_TDI-42154_Escape_char_in_couchbase_data)

**Pull request merged**